.github/workflows/cd.yml

```diff
@@ -27,25 +34,55 @@ jobs:
       uses: rlespinasse/github-slug-action@v4
 
     # Setup gcloud CLI
-    - name: Set up gcloud SDK environment
-      uses: google-github-actions/setup-gcloud@v0.4.0
+    - name: Authenticate to Google Cloud
+      id: auth
+      uses: google-github-actions/auth@v0.5.0
+      with:
+        credentials_json: ${{ secrets.GOOGLE_CREDENTIALS }}
+
+    # Setup Docker Buildx to allow use of docker cache layers from GH
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+      with:
+        driver-opts: network=host
+
+    - name: Login to Google Artifact Registry
+      uses: docker/login-action@v1.12.0
       with:
-        project_id: ${{ env.PROJECT_ID }}
-        service_account_key: ${{ secrets.GCLOUD_AUTH }}
-
-    # Build and push image to Google Container Registry
-    - name: Build
-      # Tagging w/ the commit SHA blocks the :latest tag on GCR
-      run: |
-        gcloud builds submit \
-          --config cloudbuild.yaml \
-          --substitutions SHORT_SHA="${{ env.GITHUB_SHA_SHORT }}",BRANCH_NAME="${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}"
-
-  deploy:
+        registry: us-docker.pkg.dev
+        username: _json_key
+        password: ${{ secrets.GOOGLE_CREDENTIALS }}
+
+    # Build and push image to Google Artifact Registry
+    - name: Build & push
+      id: docker_build
+      uses: docker/build-push-action@v2.8.0
+      with:
+        target: runtime
+        context: .
+        file: ./docker/Dockerfile.build
+        tags: |
+          ${{ env.GAR_BASE }}/${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}:latest
+          ${{ env.GAR_BASE }}/${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}:${{ env.GITHUB_SHA_SHORT }}
+        build-args: |
+          NETWORK=${{ github.event.inputs.network || env.NETWORK }}
+          SHORT_SHA=${{ env.GITHUB_SHA_SHORT }}
+          RUST_BACKTRACE=full
+          ZEBRA_SKIP_NETWORK_TESTS="1"
+          CHECKPOINT_SYNC=${{ github.event.inputs.checkpoint_sync || true }}
+          RUST_LOG=debug
```

Comment:
Is debug logging going to slow down `rustc` when it builds Zebra, or the Zebra tests?

Reply:
As we don't need to debug by default, and can activate it at runtime, I'll remove it as an argument.